sam_view.c

```diff
@@ -1116,6 +1158,22 @@ static bool init_state(const bam2fq_opts_t* opts, bam2fq_state_t** state_out)
     state->filetype = opts->filetype;
     state->def_qual = opts->def_qual;
 
+    state->taglist = kl_init(ktaglist);
```

Comment:
This memory is leaked, although it's a minor issue as the program is exiting.

Please consider adding a ```kl_destroy``` call at the end to tidy up.